src/domain/swaps/contracts/decoders/set-pre-signature-decoder.helper.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@nestjs/common';
+import { AbiDecoder } from '@/domain/contracts/decoders/abi-decoder.helper';
+import { getFunctionSelector, isHex, parseAbi } from 'viem';
+
+export const abi = parseAbi([
+  'function setPreSignature(bytes calldata orderUid, bool signed)',
+]);
+
+@Injectable()
+export class SetPreSignatureDecoder extends AbiDecoder<typeof abi> {
+  private readonly setPreSignatureFunctionSelector: `0x${string}`;
+
+  constructor() {
+    super(abi);
+    this.setPreSignatureFunctionSelector = getFunctionSelector(abi[0]);
+  }
+
+  /**
+   * Gets the Order UID associated with the provided transaction data.
+   *
+   * @param data - the transaction data for the setPreSignature call
+   * @returns {`0x${string}`} the order UID or null if the data does not represent a setPreSignature transaction
+   */
+  getOrderUid(data: `0x${string}`): `0x${string}` | null {
+    if (!data.startsWith(this.setPreSignatureFunctionSelector)) return null;
+    const { functionName, args } = this.decodeFunctionData({ data });
+    if (functionName !== 'setPreSignature') return null;
```

Comment:
Line 25 and 27 check the same thing here. If the former is removed, `decodeFunctionData` can throw though.

Reply:
Good call about the `throw` (honestly I think this should be documented). I will wrap it in a `try/catch` block and return `null`. The goal of this function is not to provide the caller the opportunity to deal with the error.

Reply:
Fixed in [9dca8df](https://github.com/safe-global/safe-client-gateway/pull/1239/commits/9dca8df8a3cde5031ea9fd75fc93e15fb4a30639)